lib/cherrypy/wsgiserver/wsgiserver3.py

```diff
@@ -1627,9 +1627,15 @@ def start(self):
             except:
                 pass
 
-            # So everyone can access the socket...
+            # So everyone can access the socket, this was set to 777
+            # https://docs.openstack.org/developer/bandit/plugins/set_bad_file_permissions.html
+            # This plugin test looks for the use of chmod and will alert when
+            # it is used to set particularly permissive control flags.
+            # A MEDIUM warning is generated if a file is set to group
+            # executable and a HIGH warning is reported if a file
+            # is set world writable. Warnings are given with HIGH confidence.
             try:
-                os.chmod(self.bind_addr, 0o777)
+                os.chmod(self.bind_addr, 0o755)
```

Comment:
Agree it's safer to reduce permission, but note this is a change to CherryPy. Might be a good idea to send an issue/comment to the upstream repository, and see what they think about it.

Note that in their current code, there is a `TODO` note, suggesting that they could be accepting a pull request to make that configurable https://github.com/cherrypy/cheroot/blob/6c0e8c896116f65b51678b61acd22931f6cbdf84/cheroot/server.py#L1810, but I think we won't put any efforts into that, given we are moving away from CherryPy.

Also, there is a small chance of issues in some sites, depending on the operating system and permissions for user/folder. From http://man7.org/linux/man-pages/man7/unix.7.html, under "Pathname socket ownership and permissions"

>       In the Linux implementation, pathname sockets honor the permissions
>       of the directory they are in.  Creation of a new socket fails if the
>       process does not have write and search (execute) permission on the
>       directory in which the socket is created.
>
>       On Linux, connecting to a stream socket object requires write permis‐
>       sion on that socket; sending a datagram to a datagram socket likewise
>       requires write permission on that socket.  POSIX does not make any
>       statement about the effect of the permissions on a socket file, and
>       on some systems (e.g., older BSDs), the socket permissions are
>       ignored.  Portable programs should not rely on this feature for secu‐
>       rity.

So in theory, something like 640 or 660 could work too, but there's no guarantee going from `777` to `755` or more restrictive won't affect users. So in case it gets merged, we may need to include something some where, like in the release notes I think.

Reply:
Thanks Bruno

Reply:
We should probably delete this source file. Cylc does not use this.

Reply:
Note: the bundled version of CherryPy is already heavily modified, so removing more unused-unsafe source files is more than acceptable.